bftclient/src/bft_client.cpp

```diff
@@ -98,6 +98,7 @@ Msg Client::createClientMsg(const RequestConfig& config, Msg&& request, bool rea
   header->requestLength = request.size();
   header->timeoutMilli = config.timeout.count();
   header->cidLength = config.correlation_id.size();
+  header->result = 1;  // UNKNOWN
```

Comment:
In case it is set to 1 in the ClientRequestMsgHeader, I think there is no need to set it again here.

Reply:
Yeah, I thought by default it should be UNKNOWN, but it was always taking SUCCESS. So had to change it here also.

Reply:
The basic code is written in an error-prone way. Because of:
ClientRequestMsgHeader* header = reinterpret_cast<ClientRequestMsgHeader*>(msg.data());
the header object does not get created and thus its initialization does not happen. Actually, we need to initialize here also all other header fields.